src/types/events.ts

```diff
@@ -59,3 +60,143 @@ export class BalancesTransferEvent {
         return this._chain.decodeEvent(this.event)
     }
 }
+
+export class ZenlinkProtocolAssetSwapEvent {
+    private readonly _chain: Chain
+    private readonly event: Event
+
+    constructor(ctx: EventContext)
+    constructor(ctx: ChainContext, event: Event)
+    constructor(ctx: EventContext, event?: Event) {
+        event = event || ctx.event
+        assert(event.name === 'ZenlinkProtocol.AssetSwap')
+        this._chain = ctx._chain
+        this.event = event
+    }
+
+    /**
+     *  Transact in trading \[owner, recipient, swap_path, balances\]
+     */
+    get isV902(): boolean {
```

Comment:
Do note: the bifrost implementation has more isVXXX/ getVXXX.  I believe we will need fewer implementations of this as we trade less tokens. Trimmed appropriately.